File: archive/p/php/rot13.php
<?php

if (empty($argv[1]) || $argv[1] == "" || count($argv) == 0) {
    die("Usage: please provide a string to encrypt\n");
}

function rot13(string $string) {
    return strtr($string,
        'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz',
        'NOPQRSTUVWXYZABCDEFGHIJKLMnopqrstuvwxyzabcdefghijklm');
}

echo rot13($argv[1]) . "\n";

File: archive/p/perl/rot13.pl
#!/usr/bin/env perl
use v5.42;

sub usage {
    say "Usage: please provide a string to encrypt";
    exit;
}

my ($str) = @ARGV;
usage() unless defined $str && length $str;

$str =~ tr/A-Za-z/N-ZA-Mn-za-m/;
say $str;
